src/Libraries/RevitNodes/Elements/Revision.cs

```diff
@@ -0,0 +1,261 @@
+        }
+
+        #endregion
+
+        #region Helpers for private constructors
+
+        private void InitElement(Autodesk.Revit.DB.Revision element)
+        {
+            InternalSetElement(element);
+        }
+
+        /// <summary>
+        /// Create new Revision
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="visibility"></param>
+        /// <param name="revDate"></param>
+        /// <param name="description"></param>
+        /// <param name="issued"></param>
+        /// <param name="issuedBy"></param>
+        /// <param name="issuedTo"></param>
+        /// <param name="numberType"></param>
+        private void Init(string name, RevisionVisibility visibility, string revDate, string description, bool issued, string issuedBy, string issuedTo, RevisionNumberType numberType)
+        {
+            // Get document and open transaction
+            Autodesk.Revit.DB.Document document = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+
+            // Get existing element if any
+            var RevisionElem = ElementBinder.GetElementFromTrace<Autodesk.Revit.DB.Revision>(document);
+
+            if (RevisionElem == null)
+            { 
+                RevisionElem = Autodesk.Revit.DB.Revision.Create(document); 
+            }
+            
+            // Apply properties
+            RevisionElem.Visibility = visibility;
+            RevisionElem.Description = description;
+            RevisionElem.RevisionDate = revDate;
+            RevisionElem.Issued = issued;
+            RevisionElem.IssuedBy = issuedBy;
+            RevisionElem.IssuedTo = issuedTo;
+            RevisionElem.NumberType = numberType;
+
+            InternalSetElement(RevisionElem);
+
+            // commit transaction and set element for trace
+            TransactionManager.Instance.TransactionTaskDone();
+            ElementBinder.SetElementForTrace(this.InternalElement);
+        }
+
+        #endregion
+
+        #region Public static constructors
+
+        /// <summary>
+        /// Construct a new Revit Revision by Name
+        /// </summary>
+        /// <param name="name">Revision Name</param>
+        /// <param name="visibility">Visibility settings</param>
+        /// <param name="revDate">Revision Date</param>
+        /// <param name="description">Description</param>
+        /// <param name="issued">Issuing status</param>
+        /// <param name="issuedBy">Issued by</param>
+        /// <param name="issuedTo">Issued to</param>
+        /// <param name="numberType">Number type</param>
+        /// <returns></returns>
+        public static Revision ByName(string name, string revDate, string description, bool issued, string issuedBy, string issuedTo, string visibility = "",string numberType = "")
+        {
+            RevisionVisibility revVisibility = RevisionVisibility.CloudAndTagVisible;
```

Comment:
doing a tryparse afterwards. the default is only a fallback if the tryparse fails.